m3u8/writer.go

```diff
@@ -603,6 +684,70 @@ func (p *MediaPlaylist) AppendSegment(seg *MediaSegment) error {
 	return nil
 }
 
+func (p *MediaPlaylist) AppendPartial(uri string, duration float64, independent bool) error {
+	seg := new(PartialSegment)
+	seg.URI = uri
+	seg.Duration = duration
+	seg.Independent = independent
+	return p.AppendPartialSegment(seg)
+}
+
+func (p *MediaPlaylist) AppendPartialSegment(ps *PartialSegment) error {
```

Comment:
Here the doc string is even more important, since there is a non-trivial side-effect in increasing `MaxPartIndex` so I think this should be clearly documented.